src/include/ceph_fs.h

```diff
@@ -333,18 +333,24 @@ extern const char *ceph_mds_state_name(int s);
  */
 #define CEPH_LOCK_DN          (1 << 0)
 #define CEPH_LOCK_DVERSION    (1 << 1)
-#define CEPH_LOCK_ISNAP       (1 << 4)  /* snapshot lock. MDS internal */
-#define CEPH_LOCK_IPOLICY     (1 << 5)  /* policy lock on dirs. MDS internal */
-#define CEPH_LOCK_IFILE       (1 << 6)
-#define CEPH_LOCK_INEST       (1 << 7)  /* mds internal */
-#define CEPH_LOCK_IDFT        (1 << 8)  /* dir frag tree */
-#define CEPH_LOCK_IAUTH       (1 << 9)
-#define CEPH_LOCK_ILINK       (1 << 10)
-#define CEPH_LOCK_IXATTR      (1 << 11)
-#define CEPH_LOCK_IFLOCK      (1 << 12)  /* advisory file locks */
-#define CEPH_LOCK_IVERSION    (1 << 13)  /* mds internal */
-
-#define CEPH_LOCK_IFIRST      CEPH_LOCK_ISNAP
+#define CEPH_LOCK_IQUIESCE    (1 << 4)  /* mds internal */
```

Comment:
I'm afraid that we aren't allowed to make the change in a minor release, such as the downstream we've pushed to

Reply:
maybe it's not a big deal, but we need to think whether we could have issues during an upgrade

Reply:
There are no upgrade issues. There is an incompat bit added in `mds: add quiesce set feature flag`. If a file system has more than two ranks, all must support that feature. This is enforced as part of the upgrade process:

https://docs.ceph.com/en/reef/cephfs/upgrading/#upgrading-the-mds-cluster

but also by cephadm / rook. The monitors also won't allow two MDS in the same file system to have non-equal incompat sets.

Reply:
no, the problem here is that we are updating a public header in a backward incompatible way for a minor release

Reply:
These macros are only used by the MDS. They should not be in `src/include`.

Reply:
OK in that case, it's not an issue